File: scripts/generate_report.py
import anthropic
import json
import os
from datetime import datetime
from tabulate import tabulate

# Load market data
with open('market_data.json', 'r') as f:
    data = json.load(f)

# Create market data summary
summary_parts = ["CURRENT MARKET DATA:", ""]

groups = [
    ("US MARKETS", ["S&P 500", "NASDAQ", "Dow Jones"]),
    ("ASIAN MARKETS", ["Nikkei 225", "Hang Seng", "Shanghai Composite", "ASX 200"]),
    ("EUROPEAN MARKETS", ["FTSE 100", "DAX", "CAC 40"]),
    ("CURRENCIES", ["EUR/USD", "GBP/USD", "USD/JPY"]),
    ("COMMODITIES", ["Gold", "Crude Oil"])
]

for group_name, instruments in groups:
    summary_parts.append(group_name + ":")
    for instrument in instruments:
        if instrument in data:
            info = data[instrument]
            line = "- " + instrument + ": " + str(info['price']) + " (" + str(info['change']) + "%)"
            summary_parts.append(line)
    summary_parts.append("")

market_summary = "\n".join(summary_parts)

# Generate report with Claude
print("Generating report with Claude...")
client = anthropic.Anthropic(api_key=os.environ['ANTHROPIC_API_KEY'])

prompt_text = "You are a senior financial analyst. Create a professional daily market summary using this data:\n\n"
prompt_text += market_summary
prompt_text += "\n\nInclude sections: EXECUTIVE SUMMARY, ASIAN MARKETS OVERNIGHT, EUROPEAN MARKETS SUMMARY, US MARKET OUTLOOK, CURRENCY & COMMODITIES WATCH, KEY TAKEAWAYS. Use sophisticated financial language for institutional clients."

response = client.messages.create(
    model="claude-3-5-sonnet-20241022",
    max_tokens=4000,
    messages=[{"role": "user", "content": prompt_text}]
)

# Create enhanced report file
os.makedirs('reports', exist_ok=True)
today = datetime.now().strftime('%Y-%m-%d')
filename = 'reports/market-summary-' + today + '.md'

with open(filename, 'w') as f:
    f.write("# Enhanced Daily Market Summary\n")
    current_date = datetime.now().strftime('%A, %B %d, %Y')
    f.write("**" + current_date + "**\n\n")
    f.write("*Analysis of " + str(len(data)) + " global instruments*\n\n")
    f.write("---\n\n")
    f.write(response.content[0].text)
    f.write("\n\n---\n\n")
    f.write("## MARKET DATA TABLES\n\n")
    
    for title, instruments in groups:
        table_data = []
        for instrument in instruments:
            if instrument in data:
                info = data[instrument]
                price_str = str(round(info['price'], 2))
                change_str = "+" + str(info['change']) + "%" if info['change'] >= 0 else str(info['change']) + "%"
                table_data.append([instrument, price_str, change_str])
        
        if table_data:
            f.write("### " + title + "\n\n")
            headers = ['Instrument', 'Level', 'Change %']
            table = tabulate(table_data, headers=headers, tablefmt='pipe')
            f.write(table)
            f.write("\n\n")
    
    timestamp = datetime.now().strftime('%Y-%m-%d %H:%M:%S UTC')
    f.write("\n*Generated: " + timestamp + "*\n")

# Save info for GitHub issue
with open('report_info.json', 'w') as f:
    json.dump({'filename': filename, 'count': len(data)}, f)

print("Report saved: " + filename)

File: generate-report.js
const axios = require('axios');
const fs = require('fs');
const path = require('path');

const ANTHROPIC_API_KEY = process.env.ANTHROPIC_API_KEY;
const ANTHROPIC_API_URL = 'https://api.anthropic.com/v1/messages';

const MARKET_REPORT_PROMPT = `You are a financial analyst creating a daily market summary. Please search the web for current information and create a professional report with these exact sections:

**EXECUTIVE SUMMARY**
[2-sentence overview of global market sentiment]

**ASIAN MARKETS OVERNIGHT**
Search for and report on:
- Nikkei 225, Hang Seng, Shanghai Composite, ASX 200 performance
- Major Asian corporate news or earnings
- Key economic data releases from Asia
- USD/JPY, USD/CNY, AUD/USD currency movements
- Any central bank communications from Asia
[Target: 150 words]

**EUROPEAN MARKETS SUMMARY**
Search for and report on:
- FTSE 100, DAX, CAC 40, Euro Stoxx 50 performance
- Major European corporate news
- ECB policy updates or eurozone economic data
- EUR/USD, GBP/USD movements
- Any significant political/economic developments in Europe
[Target: 150 words]

**US MARKET OUTLOOK**
Search for and report on:
- Current S&P 500, NASDAQ, DOW futures
- Key economic releases scheduled for today
- Major US earnings announcements expected
- Federal Reserve speakers or policy implications
- Any overnight developments affecting US markets
[Target: 150 words]

**KEY TAKEAWAYS**
[2-sentence summary of main trading themes for the day]

**KEY HEADLINES AND RESEARCH**
[Target: 200 words]
Summary of research published and headlines published during market closure the previous hours and any potential impacts they might have.

Use current market data from today's date. Include specific percentage moves and index levels. Write in professional financial language suitable for institutional clients.`;

async function generateMarketReport() {
    try {
        console.log('Generating market report...');
        
        const response = await axios.post(ANTHROPIC_API_URL, {
            model: 'claude-sonnet-4-20250514',
            max_tokens: 4000,
            temperature: 0.3,
            messages: [{
                role: 'user',
                content: MARKET_REPORT_PROMPT
            }]
        }, {
            headers: {
                'Content-Type': 'application/json',
                'x-api-key': ANTHROPIC_API_KEY,
                'anthropic-version': '2023-06-01'
            }
        });

        const report = response.data.content[0].text;
        
        // Create reports directory if it doesn't exist
        const reportsDir = path.join(__dirname, 'reports');
        if (!fs.existsSync(reportsDir)) {
            fs.mkdirSync(reportsDir, { recursive: true });
        }
        
        // Generate filename with current date
        const today = new Date();
        const dateStr = today.toISOString().split('T')[0]; // YYYY-MM-DD format
        const filename = `market-report-${dateStr}.md`;
        const filepath = path.join(reportsDir, filename);
        
        // Add metadata header to the report
        const reportWithMetadata = `# Daily Market Report - ${dateStr}
*Generated on: ${today.toISOString()}*

${report}

---
*This report was automatically generated using Claude AI via GitHub Actions*
`;
        
        // Write report to file
        fs.writeFileSync(filepath, reportWithMetadata);
        
        console.log(`Market report generated successfully: ${filename}`);
        console.log(`Report length: ${report.length} characters`);
        
        // Also create/update latest report for easy access
        const latestFilepath = path.join(reportsDir, 'latest-market-report.md');
        fs.writeFileSync(latestFilepath, reportWithMetadata);
        
    } catch (error) {
        console.error('Error generating market report:', error.response?.data || error.message);
        process.exit(1);
    }
}

// Run the report generation
generateMarketReport();
